### PR TITLE
Ph pod 2083

### PR DESCRIPTION
**Pull request by phendriksen100**
- Move ValidateFileStatus task inside of terra_tasks.wdl
- Updated input parameters to ValidateFileStatus
- Updated transfer and validate task to handle new inputs
- Cleaned up script to handle new inputs and added better errors

### CODE REVIEW

File: src/scripts/gdc/validate_gdc_file_status.py [hunk 1]
```diff
@@ -1,34 +1,36 @@
 import argparse
 import json
```
Comment (sahakiann) [inline src/scripts/gdc/validate_gdc_file_status.py:2]: `json` is an unused import

File: src/wdl/tasks/terra_tasks.wdl [hunk 2]
```diff
@@ -247,4 +249,35 @@ task addReadsField {
     output {
         String reads_json = read_string("reads.json")
     }
+}
+
+task ValidateFileStatus {
```
Comment (sahakiann) [inline src/wdl/tasks/terra_tasks.wdl:254]: The `ValidateFileStatus` task seems to be very specific to GDC - should this be moved to `ValidateFileStatus.wdl` instead?

File: src/wdl/tasks/terra_tasks.wdl [hunk 3]
```diff
@@ -247,4 +249,35 @@ task addReadsField {
...
+    input {
+        String program
+        String project
+        String sample_alias
+        String agg_project
+        String data_type
+        String gdc_token
+        Boolean previous_task
```
Comment (sahakiann) [inline src/wdl/tasks/terra_tasks.wdl:262]: What is the `previous_task` parameter used for?

File: src/wdl/workflows/ValidateFileStatus/ValidateFileStatus.wdl [hunk 4]
```diff
@@ -10,32 +10,38 @@ workflow ValidateGDCFileStatus {
...
-      input:
-        sample_id = sample_id,
-        file_state = file_status.file_state
+    input:
+      sample_id = sample_id,
+      file_state = file_status.file_state,
+      state = file_status.state
   }
 
   call tasks.UpsertMetadataToDataModel {
```
Comment (sahakiann) [inline src/wdl/workflows/ValidateFileStatus/ValidateFileStatus.wdl:40]: From what I can tell, this task receives a tsv that has both `state` and `file_state`. But only the `file_state` column is getting populated in the workspace metadata. Should both columns be getting populated? And if so, do users know what the difference in these two columns means?

File: src/wdl/tasks/terra_tasks.wdl (hunk 2 above)
Comment (phendriksen100) [reply src/wdl/tasks/terra_tasks.wdl:254]: So this task is actually used in the main wdl. So I moved it here so it wouldnt have to be duplicated in both

File: src/wdl/tasks/terra_tasks.wdl (hunk 3 above)
Comment (phendriksen100) [reply src/wdl/tasks/terra_tasks.wdl:262]: Its used in the main wdl, so it waits for the transfer step to finish

**Review by sahakiann — COMMENTED** [review]
I'm also seeing that some samples' file_state is not getting populated at all in the sample metadata. For example `CPT0464130004_1`'s workflow [finished successfully](https://app.terra.bio/#workspaces/gdc-submissions/CPTAC_THCA_Exomes_Ship7and12_294samples/job_history/c173f10a-8f8b-4deb-9533-e6c342d129aa), but the file state is empty in the workspace metadata.

File: src/wdl/tasks/terra_tasks.wdl (hunk 3 above)
Comment (sahakiann) [reply src/wdl/tasks/terra_tasks.wdl:262]: Ohh I missed that, ok got it!

File: src/wdl/tasks/terra_tasks.wdl (hunk 2 above)
Comment (sahakiann) [reply src/wdl/tasks/terra_tasks.wdl:254]: Makes sense, missed that thanks!

**Review by sahakiann — APPROVED** [review]

**Pull request merged**